src/sentry/monitors/logic/mark_failed.py

```diff
@@ -304,6 +326,42 @@ def create_issue_platform_occurrence(
     )
 
 
+def get_failure_reason(failed_checkins: Sequence[SimpleCheckIn]):
+    """
+    Builds a humam readible strong from a list of failed check-ins.
+
+    "3 missed check-ins detected"
+    "2 missed check-ins, 1 timeout check-in and 1 error check-in were detected"
+    "A failed check-in was detected"
+    """
+    human_names: Mapping[int, str] = dict(
+        [
+            (CheckInStatus.ERROR, "error"),
+            (CheckInStatus.MISSED, "missed"),
+            (CheckInStatus.TIMEOUT, "timeout"),
+        ]
+    )
+
+    status_counts = Counter(
+        human_names[cast(int, checkin["status"])]
```

Comment:
why do we need to cast this?

Reply:
```
"CheckInStatus" is incompatible with "int"
```

It's because

```
    human_names: Mapping[int, str] = dict(
        [
            (CheckInStatus.ERROR, "error"),
            (CheckInStatus.MISSED, "missed"),
            (CheckInStatus.TIMEOUT, "timeout"),
        ]
    )
```

I would type this as 

```
    human_names: Mapping[CheckInStatus, str] = dict(
        [
            (CheckInStatus.ERROR, "error"),
            (CheckInStatus.MISSED, "missed"),
            (CheckInStatus.TIMEOUT, "timeout"),
        ]
    )
```

But that produces

```
 Type parameter "_KT@Mapping" is invariant, but "Literal[2, 4, 5]" is not the same as "CheckInStatus"
```